src/pfe/portal/routes/projects/remoteBind.route.js

```diff
@@ -260,13 +274,32 @@ router.post('/api/v1/projects/:id/upload/end', async (req, res) => {
   }
 });
 
-function getFilesToDelete(existingFileArray, newFileArray) {
-  const filesToDeleteSet = new Set(existingFileArray);
-  newFileArray.forEach((f) => filesToDeleteSet.delete(f));
+function getPathsToDelete(existingPathsArray, newPathsArray) {
+  const filesToDeleteSet = new Set(existingPathsArray);
+  newPathsArray.forEach((f) => filesToDeleteSet.delete(f));
   return Array.from(filesToDeleteSet);
 }
 
-function deleteFilesInArray(directory, arrayOfFiles) {
+function getTopLevelDirectories(directoryArray) {
+  let topLevelDirArray = [];
+  directoryArray.forEach(dir => {
+    const existingTopLevelDirectories = topLevelDirArray.filter(rootDirectory => compareDirs(dir, rootDirectory));
+    if (existingTopLevelDirectories.length === 0) {
+      // if there are subdirectories of "dir" already in the topLevelDirArray remove them
+      topLevelDirArray = topLevelDirArray.filter(finalDir => !compareDirs(finalDir, dir));
+      topLevelDirArray.push(dir);
+    }
+  });
+  return topLevelDirArray;
+}
+
+function compareDirs(string, prefix) {
+  string = path.join('/', string, '/');
```

Comment:
I think this is trying to do something that you should probably do with `path.resolve`. Also could you call the function `isSubDir()` or similar as `compareDirs` doesn't really tell you what your comparing? (The name, the contents?)

Reply:
Path.resolve didn't achieve the same effect. I checked other Path methods and couldn't find a better method than what is already implemented, I've left it as is. Have renamed the function though.